lib/ansible/plugins/lookup/fileglob.py

```diff
@@ -15,6 +15,11 @@
       _terms:
         description: path(s) of files to read
         required: True
+      ignore_missing:
```

Comment:
Could this be a string value so people could silence warnings as well.

```yaml
missing_action:
  type: str
  choices:
  - fail
  - warn
  - ignore
  default: fail
```

I suppose the downside is `find_file_in_search_path` doesn't expose this granularity so probably not possible without even more changes.

Reply:
we also have a ignore_errors that would handle first case common to all lookukps

Reply:
but not hard to change (default warn to keep backwards compat).
```diff --git a/lib/ansible/plugins/lookup/fileglob.py b/lib/ansible/plugins/lookup/fileglob.py
index 617bc1025e..0a2c77e9c2 100644
--- a/lib/ansible/plugins/lookup/fileglob.py
+++ b/lib/ansible/plugins/lookup/fileglob.py
@@ -15,10 +15,14 @@ DOCUMENTATION = """
       _terms:
         description: path(s) of files to read
         required: True
-      ignore_missing:
-        description: Flag to control whether or not no matches found return a warning
-        type: boolean
-        default: False
+      on_missing:
+        description: Flag to control behaviour when no files are matched.
+        type: str
+        default: warn
+        choices:
+            fail: Will make no matching an Error
+            warn: Just warn that we could not match files, return empty list
+            ignore: Just return empty list, no fuss.
         version_added: "2.18"
     notes:
       - Patterns are only supported on files, not directory/paths.
@@ -56,6 +60,7 @@ RETURN = """
 import os
 import glob
 
+from ansible.errors import AnsibleError
 from ansible.plugins.lookup import LookupBase
 from ansible.module_utils.common.text.converters import to_bytes, to_text
 
@@ -70,8 +75,11 @@ class LookupModule(LookupBase):
         for term in terms:
             term_file = os.path.basename(term)
             found_paths = []
+            ignore_missing = self.get_option('on_missing') in ('ignore', 'fail')
             if term_file != term:
-                found_paths.append(self.find_file_in_search_path(variables, 'files', os.path.dirname(term), ignore_missing=self.get_option('ignore_missing')))
+                found_paths.append(self.find_file_in_search_path(variables, 'files', os.path.dirname(term), ignore_missing=ignore_missing))
+                if not found_paths and self.get_option('on_missing') == 'fail':
+                    raise AnsibleError(f"Failed to find requested path: {term}")
             else:
```